### PR TITLE
shorten sleeps in servlet tests

### PR DESCRIPTION
**Pull request by richardstartin**
We spend a lot of time sleeping in servlet tests, this PR investigates whether this time can be reduced.

### CODE REVIEW

**Comment by richardstartin** [comment]
This takes a lot of time off the servlet test cycle


Task | Duration | Result
-- | -- | --
:dd-java-agent:instrumentation:servlet:request-3 | 5m7.55s 
:dd-java-agent:instrumentation:servlet:request-3:test | 2m33.21s   
:dd-java-agent:instrumentation:servlet:request-3:testJava8Generated | 2m18.15s





Task | Duration | Result
-- | -- | --
:dd-java-agent:instrumentation:servlet:request-3 | 4m8.68s 
:dd-java-agent:instrumentation:servlet:request-3:test | 1m57.52s 
:dd-java-agent:instrumentation:servlet:request-3:testJava8Generated | 1m55.19s

File: dd-java-agent/instrumentation/servlet/request-3/src/test/groovy/TestServlet3.groovy [hunk 1]
```diff
@@ -62,7 +62,7 @@ class TestServlet3 {
       def phaser = new Phaser(2)
       def context = req.startAsync()
       if (endpoint.name().contains("TIMEOUT")) {
-        context.setTimeout(1_000)
+        context.setTimeout(100)
```
Comment (bantonsson) [inline dd-java-agent/instrumentation/servlet/request-3/src/test/groovy/TestServlet3.groovy:65]: So how much of a margin is there for hiccups on a loaded system compared to _normal_ running times?

**Review by bantonsson — APPROVED** [review]

File: dd-java-agent/instrumentation/servlet/request-3/src/test/groovy/TestServlet3.groovy (hunk 1 above)
Comment (richardstartin) [reply dd-java-agent/instrumentation/servlet/request-3/src/test/groovy/TestServlet3.groovy]: Just got a spurious timeout at 250ms, so I think I'll put the timeout back to 1000, and just skip the 2s pause

**Comment by richardstartin** [comment]
I've dropped the commit which changed the statically assigned jetty thread pool since it seemed to be interfering with jetty

**Comment by richardstartin** [comment]
Dropping the thread pool override and just skipping the 2s extra pause it's still a decent gain of over 30s






Task | Duration 
-- | -- 
:dd-java-agent:instrumentation:servlet:request-3 | 4m35.11s 
:dd-java-agent:instrumentation:servlet:request-3:test | 2m13.10s 
:dd-java-agent:instrumentation:servlet:request-3:testJava8Generated | 2m7.38s

**Review by mcculls — APPROVED** [review]

**Review by tylerbenson — APPROVED** [review]

**Pull request merged**